kiva/basecore2d.py

```diff
@@ -639,9 +640,37 @@ def arc(self, x, y, radius, start_angle, end_angle, cw=False):
         self.state.current_point = pts[-1]
 
     def arc_to(self, x1, y1, x2, y2, radius):
-        """
-        """
-        raise NotImplementedError("arc_to is not implemented")
+        """ Draw a circular arc from current point to tangent line
+
+        The arc is tangent to the line from the current point to ``(x1, y1)``,
+        and it is also tangent to the line from ``(x1, y1)`` to ``(x2, y2)``.
+        ``(x1, y1)`` is the imaginary intersection point of the two lines
+        tangent to the arc at the current point and at ``(x2, y2)``.
+
+        If the tangent point on the line from the current point to ``(x1, y1)``
+        is not equal to the current point, a line is drawn to it. Depending on
```

Comment:
I assume this is true even in the case where that line is going "backwards"; i.e., in the direction from `(x1, y1)` towards the current point?

Reply:
To clarify with an example: I'm thinking about cases like the one where the current point is `(1, 0)`, `(x1, y1)` is `(0, 0)`, `(x2, y2)` is `(0, 1)`, and the radius is `2`.